### DISCUSSION TITLE
How to do deselect records after moving from one tab to another?

### DISCUSSION THREAD

**Discussion by aryadilas** [filamentphp/filament#10731] [Help] [answered]
I'm trying to make Tab component in index page like [This](https://demo.filamentphp.com/shop/orders)  , 
If I select a record in Proccessing Tab, and then when I move to Shipped Tab, selected record in proccessing tab is still selected.
What I need is deselect record if user move to another tab.

This is my code
```
public function getTabs(): array
{
    return [
        'NonRegistered' => Tab::make(label: 'Non Registered')
            ->modifyQueryUsing(function ($query){
                return $query->doesntHave('registrations');
            }),
        'Registered' => Tab::make(label: 'Registered')
            ->modifyQueryUsing(function ($query){
                return $query->has('registrations');
            })
    ];
}
```
I put it in the list file.
I found `deselectRecordsAfterCompletion()` method but it work in Table BulkAction, is it possible to call that method inside `getTabs()` ? or maybe there is another method possible ?

**Comment by leandrocfe** [comment, marked as answer]
add this in the ListPage

```php
public function updatedActiveTab(): void
{
    $this->resetPage();
    $this->deselectAllTableRecords();
}
```

**Reply by aryadilas** [reply]
nice, thank you

**Reply by abemuisz95** [reply]
Man this is really helpful. Thanks.

On a side note, I think this need to be documented in the docs.
